unified_planning/model/problem.py

```diff
@@ -226,10 +226,38 @@ def has_fluent(self, name: str) -> bool:
                 return True
         return False
 
-    def add_fluent(self, fluent: 'up.model.fluent.Fluent', *,
-                   default_initial_value: Union['up.model.fnode.FNode', 'up.model.object.Object', bool,
-                                                int, float, Fraction] = None):
-        '''Adds the given fluent.'''
+    def add_fluent(self, fluent_or_name: Union['up.model.fluent.Fluent', str], 
+                   typename: 'up.model.types.Type' = None, *,
+                   default_initial_value: Union['up.model.fnode.FNode', 'up.model.object.Object', bool, int, float, Fraction] = None,
+                   **kwargs: 'up.model.types.Type') -> 'up.model.fluent.Fluent':
+        """Adds the given fluent to the problem.
+
+        If the first parameter is not a Fluent, the parameters will be passed to the Fluent constructor to create it.
+
+        :param fluent_or_name: Fluent instance or name of the fluent to be constructed
+        :param typename: If only the name of the fluent is given, this is the fluent's type (passed to the Fluent constructor).
+        :param default_initial_value: If provided, defines the default value taken in initial state by
+                                      a state variable of this fluent that has no explicit value.
+        :param kwargs: If only the name of the fluent is given, these are the fluent's parameters (passed to the Fluent constructor).
+        :return: The fluent passed or constructed.
+
+        Example
+        --------
+        >>> from unified_planning.shortcuts import *
+        >>> problem = Problem()
+        >>> location = UserType("Location")
+        >>> at_loc = Fluent("at_loc", BoolType(), l=location)  # creates a new fluent
+        >>> problem.add_fluent(at_loc)  # adds it to the problem
+        bool at_loc[l=Location]
+        >>> problem.add_fluent("connected", BoolType(), l1=location, l2=location)  # creates a new fluent and add it to the problem.
+        bool connected[l1=Location, l2=Location]
+        >>>
+        """
+        if isinstance(fluent_or_name, up.model.fluent.Fluent):
+            assert len(kwargs) == 0 and typename is None
```

Comment:
Here I would raise a meaningful Exception, so that the code is more user-fail-friendly